### DISCUSSION TITLE
Turning DXF into an SVG in rust.

### DISCUSSION THREAD

**Discussion by raisfeld-ori** [mozman/ezdxf#1223] [Q&A] [answered]
Hi! I need to turn a ".dxf" file into a ".svg" file in rust. I originally thought I would just need to convert every entity type into an equivalent svg object. I made [this package](https://github.com/raisfeld-ori/dxf_to_svg) that way, but it failed for a lot of layers and the colors weren't working properly. This package worked amazingly for me but I couldn't tell where the conversion of SVG to DXF happens...

I would be happy if someone were to tell me how does dxf-to-svg conversions happen in this codebase, thanks!

**Comment by mozman** [comment]
By the `drawing` add-on:

https://github.com/mozman/ezdxf/tree/master/src/ezdxf/addons/drawing

**Reply by raisfeld-ori** [reply]
that doesn't really help out... I already looked at the file, but I can't tell what am I doing wrong compared to this package

**Comment by mozman** [comment, marked as answer]
The rough workflow of the drawing add-on:

Frontend:
- resolving the DXF attributes in the context of the layout, block or viewport
- ordering entities and filtering invisible entities
- preparing the entities for the render pipeline

Render Pipeline:
- convert DXF entities (incl. text entities) to drawing primitives: lines, linear paths and filled paths
- clipping drawing primitives (viewports, clipped block references)
- resolving line types into solid line segments
- passing the drawing primitives to the Backend

Backend:
- renders the drawing primitives as SVG, PDF, PNG, JPG, HPGL2, ....
